web/src/ShareSnapshotModal.tsx

```diff
@@ -54,76 +104,64 @@ export default class ShareSnapshotModal extends PureComponent<props> {
             </p>
           )}
         </div>
-        {!hasLink && (
-          <button
-            className="ShareSnapshotModal-button"
-            onClick={this.props.handleSendSnapshot}
-          >
-            Get Link
-          </button>
-        )}
-        {hasLink && (
-          <a
-            className="ShareSnapshotModal-button"
-            href={this.props.snapshotUrl}
-            target="_blank"
-          >
-            Open
-          </a>
-        )}
-      </section>
+        {this.renderGetLinkButton()}
+      </div>
     )
   }
 
-  tiltCloudCopy() {
-    if (!this.props.tiltCloudUsername) {
+  renderGetLinkButton() {
+    const hasLink = this.props.snapshotUrl !== ""
+    if (hasLink) {
       return (
-        <section>
-          <p>
-            Connect Tilt to TiltCloud to share under your name and manage your
-            snapshots.
-          </p>
-          <form
-            className="ShareSnapshotModal-tiltCloudButtonForm"
-            action={this.props.tiltCloudSchemeHost + "/start_register_token"}
-            target="_blank"
-            method="POST"
-            onSubmit={ShareSnapshotModal.notifyTiltOfRegistration}
-          >
-            <p>You'll need a GitHub account</p>
-            <span className="ShareSnapshotModal-tiltCloudButtonForm-spacer" />
-            <input
-              className="ShareSnapshotModal-tiltCloudButton"
-              type="submit"
-              value="Connect to TiltCloud"
-            />
-            <input
-              name="token"
-              type="hidden"
-              value={cookies.get("Tilt-Token")}
-            />
-          </form>
-        </section>
+        <a
+          className="ShareSnapshotModal-button ShareSnapshotModal-button--inline"
+          href={this.props.snapshotUrl}
+          target="_blank"
+        >
+          Open
+        </a>
       )
-    } else {
-      return (
-        <p>
+    }
+    return (
+      <button
+        className="ShareSnapshotModal-button ShareSnapshotModal-button--inline"
+        onClick={this.props.handleSendSnapshot}
+      >
+        Get Link
+      </button>
+    )
+  }
+
+  renderLoginState() {
+    return (
+      <div className="ShareSnapshotModal-loginState">
+        Sharing as <strong>{this.props.tiltCloudUsername}</strong>
+      </div>
+    )
+  }
+
+  maybeRenderManageSnapshots() {
+    if (!this.props.tiltCloudUsername) {
+      return ""
```

Comment:
I think it would make more sense to return null here. Is that allowed?

Keying off of this doc: https://reactjs.org/docs/jsx-in-depth.html#booleans-null-and-undefined-are-ignored

Reply:
done